src/value.cc

```diff
@@ -511,27 +487,48 @@ std::string Value::chrString() const
   return boost::apply_visitor(chr_visitor(), this->value);
 }
 
-const Value::VectorType &Value::toVector() const
+void Value::VectorPtr::flatten() {
+  int n = 0;
+  for (unsigned int i = 0; i < (*this)->size(); i++) {
+    if ((*this)[i].type() == Value::ValueType::VECTOR) {
+      n += (*this)[i].toVectorPtr()->size();
+    } else {
+      n++;
+    }
+  }
+  Value::VectorPtr ret; ret->reserve(n);
+  for (unsigned int i = 0; i < (*this)->size(); i++) {
+    if ((*this)[i].type() == Value::ValueType::VECTOR) {
+      Value::VectorPtr &vec_ptr = (*this)[i].toVectorPtrRef();
+      for(unsigned int j = 0; j < vec_ptr->size(); ++j) {
```

Comment:
This for loop looks equivalent to `ret->insert(end(*ret), begin(vec_ptr), end(vec_ptr))`

Reply:
The `for` loop needed an additional non-const `operator *()`.

Changed to use `insert()` and move iterators via https://github.com/openscad/openscad/pull/3096/commits/ca6d46c3a9b77739395150f39ef8cd325acf2343.